internal/storage/writer/talaria/talaria.go

```diff
@@ -0,0 +1,95 @@
+package talaria
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	talaria "github.com/kelindar/talaria/client/golang"
+
+	"github.com/kelindar/talaria/internal/encoding/key"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+)
+
+// GetClient will create a Talaria client
+func GetClient(endpoint string, dialTimeout time.Duration, circuitTimeout time.Duration, maxConcurrent int, errorThresholdPercent int, nonBlocking bool) (*talaria.Client, error) {
+
+	var client *talaria.Client
+	var err error
+
+	if nonBlocking {
+		client, err = talaria.Dial(endpoint, talaria.WithNetwork(dialTimeout), talaria.WithCircuit(circuitTimeout, maxConcurrent, errorThresholdPercent), talaria.WithNonBlock())
+	} else {
+		client, err = talaria.Dial(endpoint, talaria.WithNetwork(dialTimeout), talaria.WithCircuit(circuitTimeout, maxConcurrent, errorThresholdPercent))
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// Writer to write to TalariaDB
+type Writer struct {
+	lock                  sync.Mutex
+	endpoint              string
+	dialTimeout           time.Duration
+	circuitTimeout        time.Duration
+	maxConcurrent         int
+	errorPercentThreshold int
+	nonBlocking           bool
+	client                *talaria.Client
+}
+
+// New initializes a new Talaria writer.
+func New(endpoint string, dialTimeout time.Duration, circuitTimeout time.Duration, maxConcurrent int, errorPercentThreshold int, nonBlocking bool) (*Writer, error) {
```

Comment:
looks like there's a lot of required arguments, can we make sure we have sensible defaults?

Reply:
Okay I'll follow the defaults in client and set them in writer.go